dipy/tracking/streamline.py

```diff
@@ -802,5 +878,114 @@ def values_from_volume(data, streamlines, affine=None):
+        inverse of the Mahalanobis distance, relative to the distribution of
+        coordinates at that node position across streamlines.
+    """
+    if isinstance(bundle, np.ndarray):
+        # It's an array, go with it:
+        n_points = bundle.shape[1]
+    else:
+        # It's something else, assume that it needs to be resampled:
+        bundle = np.array(set_number_of_points(bundle, n_points))
+    w = np.zeros((bundle.shape[0], n_points))
+
+    # If there's only one fiber here, it gets the entire weighting:
+    if bundle.shape[0] == 1:
+        if return_mahalnobis:
+            return np.array([np.nan])
+        else:
+            return np.array([1])
+
+    for node in range(bundle.shape[1]):
+        # This should come back as a 3D covariance matrix with the spatial
+        # variance covariance of this node across the different streamlines
+        # This is a 3-by-3 array:
+        node_coords = bundle[:, node]
+        c = np.cov(node_coords.T, ddof=0)
+        c = np.array([[c[0, 0], c[0, 1], c[0, 2]],
+                      [0, c[1, 1], c[1, 2]],
+                      [0, 0, c[2, 2]]])
+        # Calculate the mean or median of this node as well
+        # delta = node_coords - np.mean(node_coords, 0)
+        m = stat(node_coords, 0)
+        # Weights are the inverse of the Mahalanobis distance
+        for fn in range(bundle.shape[0]):
+            # calculate Mahalanobis for node on fiber[fn]
+            w[fn, node] = mahalanobis(node_coords[fn], m, np.linalg.inv(c))
+    if return_mahalnobis:
+        return w
+    # weighting is inverse to the distance (the further you are, the less you
+    # should be weighted)
+    w = 1 / w
+    # Normalize before returning, so that the weights in each node sum to 1:
+    return w / np.sum(w, 0)
+
+
+def bundle_profile(data, bundle, affine=None, n_points=100,
+                   weights=None):
+    """
+    Calculates a summarized profile of data for a bundle along its length
+
+    Parameters
+    ----------
+    data : 3D volume
+    bundle : StreamLines class instance, list of arrays, or array
+    weights : 1D array or 2D array (optional)
+        Weight each streamline (1D) or each node (2D) when calculating the
+        tract-profiles. Must sum to 1 across streamlines (in each node if
+        relevant).
+
+    Note
+    ----
+    Before providing a bundle as input to this function, you will need
+    to make sure that the streamlines in the bundle are all oriented in the
+    same orientation relative to the bundle (use :func:`orient_by_streamline`).
+    """
+    # It's already an array
+    if isinstance(bundle, np.ndarray):
+        fgarray = bundle
+    else:
+        # It's some other kind of thing (list, Streamlines, etc.).
+        # Resample each streamline to the same number of points
+        # list => np.array
+        fgarray = np.array(set_number_of_points(bundle, n_points))
+
+    values = values_from_volume(data, fgarray, affine=affine)
```

Comment:
The purpose of the resampling is to use all of the streamlines in the bundle to sample from the volume. But to do so, we need to make sure that all the streamlines have the same number of nodes, so that they roughly (!) correspond to each other. 

The approach is described in some detail in this paper: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0049790, which I will add as a reference.

Reply:
Gotcha! it is clear to me for `fgarray` 

I still think that you do not need to cast `set_number_of_points(bundle, n_points)` with  `np.array`.